src/pages/take-quiz-page/take-quiz.tsx

```diff
@@ -1,7 +1,307 @@
+import { toJS } from "mobx";
+
+import { Layout } from "../layout-page/layout";
+import { LoadingScreen } from "../../common/components/loading-screen/loading-screen";
+
+import "./take-quiz-page.scss";
+import previous_icon from "../../common/assets/previous-icon.svg";
+
+import { QUESTION_TYPES } from "../../common/constants/constant";
+import { IQuestionItem, IUserAnswersList } from "../../common/models/model";
 
 interface TakeQuizProps {}
 
 export const TakeQuiz: React.FC<TakeQuizProps> = ({}) => {
-  return <div></div>;
+  const [searchParams, setSearchParams] = useSearchParams();
+  const quizId = searchParams.get("quizId");
+
+  const navigate = useNavigate();
+
+  const [questions, setQuestions] = useState<IQuestionItem[]>([]);
+  const [questionAnswers, setQuestionAnswers] = useState<IUserAnswersList[]>(
+    []
+  );
+
+  const [quizName, setQuizName] = useState("");
+
+  const [index, setIndex] = useState(0);
+
+  const [loading, setLoading] = useState(true);
+
+  const changeCheckedAnswer = (question: any, indexAns: number) => {
+    if (
+      QuizStore.userAnswers[index]?.questionType === QUESTION_TYPES.SINGLE.value
+    ) {
+      setQuestionAnswers((currentState) => {
+        let arrayCopy = [...currentState];
+        arrayCopy = arrayCopy.map((answer, i) => {
+          if (i === indexAns) {
+            return {
+              ...answer,
+              isCorrectAnswer: !answer.isCorrectAnswer,
+            };
+          } else {
+            return {
+              ...answer,
+              isCorrectAnswer: false,
+            };
+          }
+        });
+
+        QuizStore.editUserQuestionAnswer(question.id, {
+          ...question,
+          id: question.id,
+          answerList: [...arrayCopy],
+        });
+        return arrayCopy;
+      });
+    } else if (
+      questions[index]?.question?.questionType === QUESTION_TYPES.MULTIPLE.value
+    ) {
+      setQuestionAnswers((currentState) => {
+        let arrayCopy = [...currentState];
+        arrayCopy[indexAns].isCorrectAnswer =
+          !arrayCopy[indexAns].isCorrectAnswer;
+
+        QuizStore.editUserQuestionAnswer(question.id, {
+          ...question,
+          id: question.id,
+          answerList: [...arrayCopy],
+        });
+        return arrayCopy;
+      });
+    }
+  };
+
+  function changeElement(type: string) {
+    if (type === "next") {
+      setIndex(index + 1);
+      QuizStore.selectActiveQuestion(QuizStore.userAnswers[index + 1]?.id);
+    } else {
+      setIndex(index - 1);
+      QuizStore.selectActiveQuestion(QuizStore.userAnswers[index - 1]?.id);
+    }
+  }
+
+  const calculateFinalScore = () => {
+    let finalScore = 0;
+    questions.forEach((question, indexQ) => {
+      // Verify if the array of correct answers is equal with the user's answers
+      if (
+        JSON.stringify(question?.question?.answerList) ===
+        JSON.stringify(QuizStore.userAnswers[indexQ].answerList)
```

Comment:
I suggest checking each element separately to see if the current answer is actually selected. JSON.stringify is not the most reliable method as the contents of the array could differ and would result in a false statement.